libsql-sqlite3/src/main.c

```diff
@@ -3655,36 +3647,25 @@ int sqlite3_open(
   sqlite3 **ppDb
 ){
   return openDatabase(zFilename, ppDb,
-                      SQLITE_OPEN_READWRITE | SQLITE_OPEN_CREATE, NULL, NULL, NULL);
+                      SQLITE_OPEN_READWRITE | SQLITE_OPEN_CREATE, NULL, sqlite3_create_wal);
 }
 int sqlite3_open_v2(
   const char *filename,   /* Database filename (UTF-8) */
   sqlite3 **ppDb,         /* OUT: SQLite db handle */
   int flags,              /* Flags */
   const char *zVfs        /* Name of VFS module to use */
 ){
-  return openDatabase(filename, ppDb, (unsigned int)flags, zVfs, NULL, NULL);
+  return openDatabase(filename, ppDb, (unsigned int)flags, zVfs, sqlite3_create_wal);
 }
 
 int libsql_open(
   const char *filename,   /* Database filename (UTF-8) */
   sqlite3 **ppDb,         /* OUT: SQLite db handle */
   int flags,              /* Flags */
   const char *zVfs,       /* Name of VFS module to use, NULL for default */
-  const char *zWal        /* Name of WAL module to use */
-) {
-  return openDatabase(filename, ppDb, (unsigned int)flags, zVfs, zWal, NULL);
-}
-
-int libsql_open_v2(
-  const char *filename,   /* Database filename (UTF-8) */
-  sqlite3 **ppDb,         /* OUT: SQLite db handle */
-  int flags,              /* Flags */
-  const char *zVfs,       /* Name of VFS module to use, NULL for default */
-  const char *zWal,       /* Name of WAL module to use */
-  void* pWalMethodsData   /* User data, passed to the libsql_wal struct*/
+  libsql_create_wal create_wal   /* create_wal instance, in charge of instanciating a wal */
```

Comment:
That's going to be very tricky with backward compatibility, because old user code will pass a `const char *` here, and we'll try to interpret it as `libsql_create_wal` (which should be renamed, as in the main review message), and then we crash.

For the sake of all clients, I suggest the following:
1. Leave `libsql_open` and `libsql_open_v2` as is, and error out on non-NULL `zWal`
2. Add `libsql_open_v3` with your new interface.